### PR TITLE
[MODULAR] [READY] Makes Skirtle/Turtleneck fully cosmetic and gives it sensors

### PR DESCRIPTION
**Pull request by lyricalpaws**
## About The Pull Request
For some reason, the tacticool turtleneck is a child of the tactical turtleneck. It removes the armour, but didn't get sensors. This PR makes a new child of the the clothing with sensors, and removes the high fire and acid armour, making the tacticool turtleneck and skirtleneck on par with a regular jumpsuit from loadout. Also adds syndicate and nt coloured (red and blue) turtlenecks. This is to distinguish the uplink version from the loadout one more.

Do note this is my first proper PR, if any of the changes aren't compliant with standard please notify me and I'll update it asap.

## Why It's Good For The Game
Loadout jumpsuits shouldn't be having armour and a lack of sensors.

## Changelog
:cl:
tweak: Removes tacticool armour
fix: Gives tacticool sensors
add: New turtleneck sprites
/:cl:

## Credits
@RatFromTheJungle for providing the sprites for the turtlenecks

### CODE REVIEW

File: code/modules/clothing/under/syndicate.dm [hunk 1]
```diff
@@ -41,14 +41,18 @@
 	desc = "Just looking at it makes you want to buy an SKS, go into the woods, and -operate-."
 	icon_state = "tactifool"
 	inhand_icon_state = "bl_suit"
-	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
+	has_sensor = HAS_SENSORS //SKYRAT EDIT
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	has_sensor = HAS_SENSORS //SKYRAT EDIT - ADDITION
```

File: code/modules/clothing/under/syndicate.dm [hunk 2]
```diff
@@ -41,14 +41,18 @@
...
+	//armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	//armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40) //SKYRAT EDIT - REMOVAL
```

File: code/modules/clothing/under/syndicate.dm [hunk 3]
```diff
@@ -41,14 +41,18 @@
...
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0) //SKYRAT EDIT
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0) //SKYRAT EDIT - ADDITION
```

File: code/modules/clothing/under/syndicate.dm [hunk 4]
```diff
@@ -41,14 +41,18 @@
...
 
 /obj/item/clothing/under/syndicate/tacticool/skirt
 	name = "tacticool skirtleneck"
 	desc = "Just looking at it makes you want to buy an SKS, go into the woods, and -operate-."
 	icon_state = "tactifool_skirt"
 	inhand_icon_state = "bl_suit"
-	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
+	has_sensor = HAS_SENSORS //SKYRAT EDIT
+	//armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	//armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40) //SKYRAT EDIT - REMOVAL
```

File: code/modules/clothing/under/syndicate.dm [hunk 5]
```diff
@@ -41,14 +41,18 @@
...
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0) //SKYRAT EDIT
 
 /obj/item/clothing/under/syndicate/tacticool/skirt
 	name = "tacticool skirtleneck"
 	desc = "Just looking at it makes you want to buy an SKS, go into the woods, and -operate-."
 	icon_state = "tactifool_skirt"
 	inhand_icon_state = "bl_suit"
-	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
+	has_sensor = HAS_SENSORS //SKYRAT EDIT
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	has_sensor = HAS_SENSORS //SKYRAT EDIT - ADDITION
```

File: code/modules/clothing/under/syndicate.dm [hunk 6]
```diff
@@ -41,14 +41,18 @@
...
+	//armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 50, ACID = 40)
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0) // SKYRAT EDIT
```
Comment (Swiftfeather) [inline code/modules/clothing/under/syndicate.dm]:
```suggestion
	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0) // SKYRAT EDIT - ADDITION
```

**Review by Swiftfeather — CHANGES_REQUESTED** [review]

**Comment by SpaceVampire** [comment]
I would honestly rather just remove it from the load-out system, instead of changing the actual values.
If we think that having some Acid and Fire armor is that much of an issue.
Adding sensors to it also removes a lot of the appeal.

**Comment by lyricalpaws** [comment]
@Swiftfeather Thanks for that ^^

**Comment by Swiftfeather** [comment]
> Adding sensors to it also removes a lot of the appeal.

Just turn them off

**Comment by lyricalpaws** [comment]
> 
> 
> I would honestly rather just remove it from the load-out system, instead of changing the actual values.
> If we think that having some Acid and Fire armor is that much of an issue.
> Adding sensors to it also removes a lot of the appeal.

While it's not much of an issue, I find it weird that there's that much armour for something you can get from an arcade machine. That and I do enjoy wearing the tacticool stuff roundstart.

**Comment by SpaceVampire** [comment]
> > Adding sensors to it also removes a lot of the appeal.
> 
> Just turn them off

It is possible for them to flick back on by mistake.
It's part of the whole theme that it doesn't have em.
And while I do like the load-out system, I feel like we added too much stuff at points.
Where the Turtleneck was supposed to be something you had to seek out.
And that is why it was allowed to have some extra features to it.

**Comment by SpaceVampire** [comment]
Also. Correct me if I am wrong.
But Acid and Fire armor don't really do all that much.
I mean, sure, its better then not having anything.
But that was kind of balanced out by the whole suit sensors thing.

**Comment by Funce** [comment]
Please discuss the balance or necessity of it elsewhere.

**Comment by lyricalpaws** [comment]
Fucking mad man grudge coded his own fix

**Comment by hugmeorelse** [comment]
Sure, if the actual antagonist items are still strong, this is fine.

**Comment by hugmeorelse** [comment]
Sure, if the actual antagonist items are still strong, this is fine.

If we want to honor @SpaceVampire's concern, why not make a variant that does have one like a Mk. II that gives those benefits but has to be scavenged?

**Comment by lyricalpaws** [comment]
> 
> 
> Sure, if the actual antagonist items are still strong, this is fine.
> 
> If we want to honor @SpaceVampire's concern, why not make a variant that does have one like a Mk. II that gives those benefits but has to be scavenged?

I haven't touched the antagonist suit, and don't plan to. I've only touched the loadout ones. And I'm reluctant to honor anything of SpaceVampire thusfar, as they've only been rude to me in my experience. If they want to make a mk2 design, then I have no objections to this.

**Comment by hugmeorelse** [comment]
> > Sure, if the actual antagonist items are still strong, this is fine.
> > If we want to honor @SpaceVampire's concern, why not make a variant that does have one like a Mk. II that gives those benefits but has to be scavenged?
> 
> I haven't touched the antagonist suit, and don't plan to. I've only touched the loadout ones. And I'm reluctant to honor anything of SpaceVampire thusfar, as they've only been rude to me in my experience. If they want to make a mk2 design, then I have no objections to this.

Stop creating grudges where it doesn't need to exist. I'll talk to them myself, but don't add drama where there doesn't need to be any. People need to stop acting like children over this.

**Comment by YakumoChen** [comment]
I thought the whole point was that it made you look like an antagonist and had no sensors as part of the theme / as a punishment for looking like an antag? 

You might as well make the scavenged item the actual tactical jumpsuit at that rate if you're going to make that suggestion

**Comment by YakumoChen** [comment]
Fully modularize this by making a child item and making it the loadout item instead of editing the values in base code.

**Comment by lyricalpaws** [comment]
> 
> 
> Fully modularize this by making a child item and making it the loadout item instead of editing the values in base code.

I'm not sure what you mean by this? The tacticool turtleneck is a child of the tactical turtleneck. And this doesn't edit the values of the tactical one, just the tacticool one.

**Comment by YakumoChen** [comment]
Make a loadout ready item in modular code that is a child of tactical turtleneck, with no armor values and sensors, and use that for the loadout instead of the existing items.

**Comment by lyricalpaws** [comment]
> 
> 
> Make a loadout ready item in modular code that is a child of tactical turtleneck, with no armor values and sensors, and use that for the loadout instead of the existing items.

With all due respect, it's a lot less work to just edit the base tacticool outfit and annotate it than to make an entirely new module for what? 2 items that had 2 changes each?

And the code isn't any worse off from this either

**Comment by YakumoChen** [comment]
NonModularization is supposed to be the exception, not the rule. Loadout is already modular, theres no need to make exceptions for it. You dont need to make an entirely new module when the loadout module exists either, so that's no excuse.

**Comment by lyricalpaws** [comment]
You'd think the module would be called "loadout" or something but apparently not

**Comment by lyricalpaws** [comment]
image
It's already in the loadout module if you mean a reference, as for literally every other item, they're elsewhere in a not modular folder.

**Comment by YakumoChen** [comment]
You would need to write a child item in Skyrat-tg/modular_skyrat/modules/customization/modules/clothing/under/misc.dm as obj/item_clothing/under/syndicate/[childname] or syndicate/tacticool/sensors, add the new vars and a name/desc if you want and change the item path in the loadout file to the new item.

**Comment by YakumoChen** [comment]
> > Make a loadout ready item in modular code that is a child of tactical turtleneck, with no armor values and sensors, and use that for the loadout instead of the existing items.
> 
> With all due respect, it's a lot less work to just edit the base tacticool outfit and annotate it than to make an entirely new module for what? 2 items that had 2 changes each?
> 
> And the code isn't any worse off from this either

Modularizarion is for the maintainers to save headaches, it should be done whenever possible, no matter how small the change

**Comment by lyricalpaws** [comment]
> 
> 
> > > Make a loadout ready item in modular code that is a child of tactical turtleneck, with no armor values and sensors, and use that for the loadout instead of the existing items.
> > 
> > 
> > With all due respect, it's a lot less work to just edit the base tacticool outfit and annotate it than to make an entirely new module for what? 2 items that had 2 changes each?
> > And the code isn't any worse off from this either
> 
> Modularizarion is for the maintainers to save headaches, it should be done whenever possible, no matter how small the change

Trust me, I've seen unmodular code, I worked on hyperstation for the longest time.

**Review by ZephyrTFA — APPROVED** [review]

**Comment by thestubborn** [comment]
show off the new sprites

**Comment by lyricalpaws** [comment]
image
NT Turtleneck

image
NT Skirtleneck

image
Syndie Turtleneck

image
Syndie Skirtleneck

Reminder that these are sprites done by RatFromTheJungle and not myself, so any criticism or praise goes to them.

**Comment by lyricalpaws** [comment]
I'd show them off on characters, but I don't have the time or effort to at this moment, I'm just about to crash and sleep.

**Comment by thestubborn** [comment]
> 
> 
> I'd show them off on characters, but I don't have the time or effort to at this moment, I'm just about to crash and sleep.

its fine, my prs just screenshot the sprites
they're good

**Comment by lyricalpaws** [comment]
Gonna move the syndie ones to the skyrat folder later when I have time

**Comment by lyricalpaws** [comment]
Modularised the red outfit now, had to edit a bit of the syndicate code, just to make sure that they still pull from the right place, since the base outfit doesn't.

**Review by ZephyrTFA — APPROVED** [review]
It's not the preferred method to override icons but it follows the guidelines and it likely works flawlessly. 
LGTM

**Comment by lyricalpaws** [comment]
> 
> 
> It's not the preferred method to override icons but it follows the guidelines and it likely works flawlessly.
> LGTM

Couldn't think of a graceful way to do it without making more children

File: code/modules/clothing/under/syndicate.dm [hunk 7]
```diff
@@ -67,6 +70,7 @@
 	icon_state = "camogreen"
 	inhand_icon_state = "g_suit"
 	can_adjust = FALSE
+	worn_icon = 'icons/mob/clothing/under/syndicate.dmi' //SKYRAT EDIT - ADDITION
```
Comment (thestubborn) [inline code/modules/clothing/under/syndicate.dm]: couldnt you just do what #6450 done but with worn icon?
and point the icons/mob/clothing/under/syndicate to uniform dmi in our modular stuff and make this fully modular?

Comment (lyricalpaws) [reply code/modules/clothing/under/syndicate.dm]: I can try to, yeah.

File: modular_skyrat/modules/customization/modules/clothing/under/costumes.dm [hunk 8]
```diff
@@ -96,22 +96,43 @@
...
 	has_sensor = HAS_SENSORS
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 0, ACID = 0)
 	fitted = FEMALE_UNIFORM_TOP
 	dying_key = DYE_REGISTRY_JUMPSKIRT
+
+/obj/item/clothing/under/syndicate/tacticool/CtrlShiftClick(mob/user) //This handles changing the design between new and old for skirtle and turtlenecks
+	. = ..()
+	if(user.canUseTopic(src, TRUE, FALSE, TRUE, TRUE, FALSE))
+		var/style = "tactifool"
+		var/choice = input(user, "Choose the a reskin for [src]","Reskin Outfit") as null|anything in list("Black (Original)", "Navy (New)")
```
Comment (ZephyrTFA) [inline modular_skyrat/modules/customization/modules/clothing/under/costumes.dm]:
```suggestion
		var/choice = tgui_alert(user, "Choose the a reskin for [src]", "Reskin Outfit", list("Black (Original)", "Navy (New)"))
```

**Comment by lyricalpaws** [comment]
we love merge conflicts with dmi files

**Comment by lyricalpaws** [comment]
That should have fixed the conflict, looking at the icon diff

**Comment by lyricalpaws** [comment]
That fully fixes it, was missing blue and red

**Pull request merged**